### PR TITLE
Bug 1908343: Allow clicking on column labels to configure column management in dialog

### PR DESCRIPTION
**Pull request by jcaianirh**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1908343

columnchoolser

cc: @gdoyle1

### CODE REVIEW

**Review by jhadvig — COMMENTED** [review]
Im a bit puzzled regarding this change, not sure why the `column.id` is not mandatory, it could cause behaviour issues here. Added two comments/questions.

Otherwise the changes is looking good

File: frontend/public/components/modals/column-management-modal.tsx [hunk 1]
```diff
@@ -26,35 +26,40 @@ const readOnlyColumns = new Set([NAME_COLUMN_ID]);
...
+  const isColumnIdDisabled = (id: string) => {
+    return (disableUncheckedRow && !checkedColumns.has(id)) || readOnlyColumns.has(id);
+  };
+
+  return (
+    <DataListItem
+      aria-labelledby={`table-column-management-item-${column.id}`}
+      key={column.id}
+      className="pf-c-data-list__item--transparent-bg"
+      id={isColumnIdDisabled(column.id) ? '' : column.id}
```
[the remark this answers is not in the document]
Comment (jcaianirh) [reply frontend/public/components/modals/column-management-modal.tsx:41]: @jhadvig key is unique, I just added id, and that gets passed into the handler , but I have an idea to maybe restructure a little based on your comment.

[the remark this answers is not in the document]
Comment (jhadvig) [reply frontend/public/components/modals/column-management-modal.tsx:41]: @jcaianirh ok, please let me know when it's ready for another review :)

File: frontend/public/components/modals/column-management-modal.tsx [hunk 2]
```diff
@@ -26,35 +26,40 @@ const readOnlyColumns = new Set([NAME_COLUMN_ID]);
...
+      onClick={onClick}
+    >
+      <DataListItemRow>
+        <DataListCheck
+          isDisabled={isColumnIdDisabled(column.id)}
+          aria-labelledby={`table-column-management-item-${column.id}`}
```
[the remark this answers is not in the document]
Comment (jcaianirh) [reply frontend/public/components/modals/column-management-modal.tsx:47]: @jhadvig used consistent id and only use onClick when column is enabled.

File: frontend/public/components/modals/column-management-modal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcaianirh) [reply frontend/public/components/modals/column-management-modal.tsx]: @jhadvig ready for another review, kept id consistent and only use onClick when the column is not disabled.

**Review by gdoyle1 — APPROVED** [review]
LGTM! @jcaianirh

File: frontend/public/components/modals/column-management-modal.tsx [hunk 3]
```diff
@@ -26,35 +26,40 @@ const readOnlyColumns = new Set([NAME_COLUMN_ID]);
...
-    <DataListItemRow>
-      <DataListCheck
-        isDisabled={
-          (disableUncheckedRow && !checkedColumns.has(column.id)) || readOnlyColumns.has(column.id)
-        }
-        aria-labelledby={`table-column-management-item-${column.id}`}
-        checked={checkedColumns.has(column.id)}
-        name={column.title}
-        id={column.id}
-        onChange={onChange}
```
Comment (spadgett) [inline frontend/public/components/modals/column-management-modal.tsx:46]: Does removing `onChange` break selection with a keyboard or screenreader?

**Review by spadgett — COMMENTED** [review]
Should we look at this as an upstream Patternfly accessibility issue? It feels like we're causing ourselves problems by having a checkbox that is not a `checkbox` element.

**Comment by jcaianirh** [comment]
@spadgett discussing this with the pf team and @jessiehuff to fix both the pf demo and this cm code. I've also opened an issue in pf here: https://github.com/patternfly/patternfly-react/issues/5273

**Comment by spadgett** [comment]
/override ci/prow/ceph-storage-plugin

The ceph-storage-plugin job should only have run on PRs with changes to `frontend/packages/ceph-storage-plugin`. It failed due to an unrelated CI config change. Overriding status on this PR.

File: frontend/public/components/modals/column-management-modal.tsx (hunk 3 above)
Comment (jcaianirh) [reply frontend/public/components/modals/column-management-modal.tsx]: @spadgett simplified solution, and kept onChange

**Comment by jcaianirh** [comment]
@spadgett simplified this solution and ran it by @jessiehuff for accessibility verification.

File: frontend/public/components/modals/column-management-modal.tsx [hunk 4]
```diff
@@ -48,7 +48,9 @@ const DataListRow: React.FC<DataListRowProps> = ({
       <DataListItemCells
         dataListCells={[
           <DataListCell id={`table-column-management-item-${column.id}`} key={column.id}>
-            {column.title}
+            <label className="co-label--plain" htmlFor={column.id}>
```
Comment (spadgett) [inline frontend/public/components/modals/column-management-modal.tsx:51]: I think this is better, but I'm a little worried that it makes assumptions about the markup Patternfly generates (that the checkbox will have this `id`). 

I feel like `DataList` is the wrong component. We're using it for checkboxs, but it's really meant to be more of a table with selectable rows. That's why we're having accessibility problems.

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**